### ISSUE TITLE
Support for Pivot and Unpivot the columns in a any given dataset

### ISSUE THREAD

**Issue by 007ramukakaji** [opensearch-project/sql#689]
As opensearch sql supports almost all joins and functions like date, trigonometry, mathematical etc.
We need important function in dataset that is Pivot and Unpivot

A special String like PIVOT and UNPIVOT with approach similar to Oracle database or MSSQL Database should do the job

Please let me know if this feature is already existing, if not it can be a very important feature for the next release

**Comment by dai-chen** [comment]
@007ramukakaji Thanks for reporting the issue! I assume you're talking about PIVOT function, ex. in SQLServer, which rotates table. If so, it's missing in our query engine. Personally, I think we should prioritize these kinds of analytics functions.

**Comment by 007ramukakaji** [comment]
@dai-chen yes exactly the feature for PIVOT and UNPIVOT function at sql level just like we have in SQL Server

**Comment by dai-chen** [comment]
> @dai-chen yes exactly the feature for PIVOT and UNPIVOT function at sql level just like we have in SQL Server

Thanks for the info! We will evaluate the efforts required. For now, I'm afraid you have to use standard SQL conditional clause, ex. IF or CASE, to "pivot" data manually. Here is the doc for your reference: https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#case

**Comment by dai-chen** [comment]
"Pivoting in SQL" section talks about how it works: https://www.databricks.com/blog/2018/11/01/sql-pivot-converting-rows-to-columns.html. It can be done by adding new Pivot operator or probably just transforming to equivalent `SELECT CASE ... GROUP BY ...`. Need further research.

**Comment by 007ramukakaji** [comment]
> > @dai-chen yes exactly the feature for PIVOT and UNPIVOT function at sql level just like we have in SQL Server
> 
> Thanks for the info! We will evaluate the efforts required. For now, I'm afraid you have to use standard SQL conditional clause, ex. IF or CASE, to "pivot" data manually. Here is the doc for your reference: https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#case

@dai-chen  this one is great for PIVOT, how can i do for UNPIVOT from Columns to rows, you have any references
